core/capabilities/ccip/ccipsolana/extradatadecoder.go

```diff
@@ -26,6 +29,48 @@ var (
 	evmExtraArgsV2Tag = hexutil.MustDecode("0x181dcf10")
 )
 
+type ExtraDataCodec struct{}
+
+func NewExtraDataCodec() ExtraDataCodec {
+	return ExtraDataCodec{}
+}
+
+func (c ExtraDataCodec) DecodeExtraArgs(extraArgs cciptypes.Bytes, sourceChainSelector cciptypes.ChainSelector) (map[string]any, error) {
+	if len(extraArgs) == 0 {
+		// return empty map if extraArgs is empty
+		return nil, nil
+	}
+
+	family, err := chainsel.GetSelectorFamily(uint64(sourceChainSelector))
+	if err != nil {
+		return nil, fmt.Errorf("failed to get chain family for selector %d: %w", sourceChainSelector, err)
+	}
+
+	if family != chainsel.FamilySolana {
+		return nil, fmt.Errorf("chain selector family is %d, expect %s", sourceChainSelector, chainsel.FamilySolana)
+	}
+
+	return DecodeExtraArgsToMap(extraArgs)
+}
+
+func (c ExtraDataCodec) DecodeTokenAmountDestExecData(destExecData cciptypes.Bytes, sourceChainSelector cciptypes.ChainSelector) (map[string]any, error) {
+	if len(destExecData) == 0 {
+		// return empty map if destExecData is empty
+		return nil, nil
+	}
+
+	family, err := chainsel.GetSelectorFamily(uint64(sourceChainSelector))
+	if err != nil {
+		return nil, fmt.Errorf("failed to get chain family for selector %d: %w", sourceChainSelector, err)
+	}
```

Comment:
All of these chain family checks are now unnecessary since they happen inside the oracle creator